pyproject.toml

```diff
@@ -33,39 +33,39 @@ classifiers = [
 ]
 dependencies = [
   # Core generation dependencies, pinned for reproducible builds.
-  "accelerate==1.0.1",
-  "bitsandbytes==0.45.0; sys_platform!='darwin'",
+  "accelerate",
+  "bitsandbytes; sys_platform!='darwin'",
   "clip_anytorch==2.6.0",       # replacing "clip @ https://github.com/openai/CLIP/archive/eaa22acb90a5876642d0507623e859909230a52d.zip",
   "compel==2.0.2",
-  "controlnet-aux==0.0.7",
-  "diffusers[torch]==0.31.0",
+  "controlnet-aux",
+  "diffusers[torch]",
   "gguf==0.10.0",
   "invisible-watermark==0.2.0", # needed to install SDXL base and refiner using their repo_ids
   "mediapipe==0.10.14",         # needed for "mediapipeface" controlnet model
   "numpy<2.0.0",
   "onnx==1.16.1",
   "onnxruntime==1.19.2",
   "opencv-python==4.9.0.80",
-  "pytorch-lightning==2.1.3",
-  "safetensors==0.4.3",
+  "pytorch-lightning",
+  "safetensors",
   # sentencepiece is required to load T5TokenizerFast (used by FLUX).
   "sentencepiece==0.2.0",
   "spandrel==0.3.4",
-  "timm==0.6.13",               # needed to override timm latest in controlnet_aux, see  https://github.com/isl-org/ZoeDepth/issues/26
-  "torch<2.5.0",                # torch and related dependencies are loosely pinned, will respect requirement of `diffusers[torch]`
+  "torch~=2.6.0",                # torch and related dependencies are loosely pinned, will respect requirement of `diffusers[torch]`
   "torchmetrics",
   "torchsde",
   "torchvision",
-  "transformers==4.46.3",
+  "timm~=1.0.0",                 # controlnet-aux depends on a version of timm that breaks LLaVA. explicitly pinning `timm` here, which results in a downgrade of `controlnet-aux`. https://github.com/huggingface/controlnet_aux/pull/101. If this poses a problem, we need to decide between newer `controlnet_aux` and LLaVA, OR we can fork `controlnet_aux` and update the pin.
+  "transformers",
 
   # Core application dependencies, pinned for reproducible builds.
-  "fastapi-events==0.11.1",
-  "fastapi==0.111.0",
-  "huggingface-hub==0.26.1",
-  "pydantic-settings==2.2.1",
-  "pydantic==2.7.2",
-  "python-socketio==5.11.1",
-  "uvicorn[standard]==0.28.0",
+  "fastapi-events",
+  "fastapi",
+  "huggingface-hub",
+  "pydantic-settings",
+  "pydantic<2.9",
```

Comment:
Pydantic is currently at v2.10.6 stable, any reason in particular to pin it below that?

Reply:
yes - the issue we saw with "unbound input" (i can't recall the exact wording) warnings in the console when debugging the LLaVA node were caused by pydantic > 2.9. I did not dig into troubleshooting that, but it would be the next thing to address.

Reply:
I've fixed the issue caused by the pydantic version bump and updated this pr accordingly